Automattic/wp-calypso
# Billing: Replace AddPaymentMethod by ChangePaymentMethod at site level

#### Changes proposed in this Pull Request

* We have the awkward situation that if a plan was purchased with a non-card method (like credits or paypal) then on the billing management page, you only have the option to add a new card when changing the payment method. This PR replaces the content of the `.../payment-method/add` route with the `ChangePaymentMethod` components, allowing the user to add a stored card in this flow (or paypal, once we add that option). It also updates the link text and page titles to show "Add Payment Method" precisely when the original purchase used credits, and "Change Payment Method" when there is a non-credits method.

This also deletes the old behavior of `ChangePaymentMethod`, so it should not be deployed until after the new behavior is no longer gated behind the `purchases/new-payment-methods` flag.

#### Testing instructions

- On a site purchased with credits:
    - Navigate to the billing page at `Plans > Billing`. Verify that the "Payment Method" is reported as credits and that you see "Add Payment Method" among the management links. Click to add a payment method.
    - Verify that the page title is "Add Payment Method", that your list of stored cards loads (if there are any), and that the first method is preselected.
- On a site purchased with paypal:
    - Navigate to the billing page at `Plans > Billing`. Verify that the payment method is reported as paypal and that you see "Change Payment Method" among the management links. Click to change the payment method.
    - Verify that the page title is "Change Payment Method" and that the first method in the list is preselected.
- On a site purchased with a card:
    - Navigate to the billing page at `Plans > Billing`. Verify that the payment method is reported as a card (it will just show the card brand and last 4) and that you see "Change Payment Method" among the management links. Click to change the payment method.
    - Verify that the page title is "Change Payment Method" and that your currently assigned card is preselected.

Fixes: https://github.com/Automattic/wp-calypso/issues/48457
